src/EditorFeatures/Test2/GoToDefinition/GoToDefinitionTests.vb

```diff
@@ -3486,5 +3493,1398 @@ class Test
+        Public Sub TestVisualBasicGoToOnContinueFor_Continue()
+            Dim workspace =
+<Workspace>
+    <Project Language="Visual Basic" CommonReferences="true">
+        <Document>
+Class C
+    Sub M()
+        [||]For index As Integer = 1 To 5
+             Continue$$ For
+        Next
+    End Sub
+End Class
+        </Document>
+    </Project>
+</Workspace>
+
+            Test(workspace)
+        End Sub
+
+        <WpfFact, Trait(Traits.Feature, Traits.Features.GoToDefinition)>
+        Public Sub TestVisualBasicGoToOnContinueFor_For()
+            Dim workspace =
+<Workspace>
+    <Project Language="Visual Basic" CommonReferences="true">
+        <Document>
+Class C
+    Sub M()
+        [||]For index As Integer = 1 To 5
+             Continue For$$
+        Next
+    End Sub
+End Class
+        </Document>
+    </Project>
+</Workspace>
+
+            Test(workspace)
+        End Sub
+
+        <WpfFact, Trait(Traits.Feature, Traits.Features.GoToDefinition)>
+        Public Sub TestVisualBasicGoToOnExitFor_For()
+            Dim workspace =
+<Workspace>
+    <Project Language="Visual Basic" CommonReferences="true">
+        <Document>
+Class C
+    Sub M()
+        For index As Integer = 1 To 5
+             Exit For$$
+        Next[||]
+    End Sub
+End Class
+        </Document>
+    </Project>
+</Workspace>
+
+            Test(workspace)
+        End Sub
+
+        <WpfFact, Trait(Traits.Feature, Traits.Features.GoToDefinition)>
+        Public Sub TestVisualBasicGoToOnContinueForEach_For()
+            Dim workspace =
+<Workspace>
+    <Project Language="Visual Basic" CommonReferences="true">
+        <Document>
+Class C
+    Sub M()
+        [||]For Each element In Nothing
+             Continue For$$
+        Next
+    End Sub
+End Class
+        </Document>
+    </Project>
+</Workspace>
+
+            Test(workspace)
+        End Sub
+
+        <WpfFact, Trait(Traits.Feature, Traits.Features.GoToDefinition)>
+        Public Sub TestVisualBasicGoToOnExitForEach_For()
+            Dim workspace =
+<Workspace>
+    <Project Language="Visual Basic" CommonReferences="true">
+        <Document>
+Class C
+    Sub M()
+        For Each element In Nothing
+             Exit For$$
+        Next[||]
+    End Sub
+End Class
+        </Document>
+    </Project>
+</Workspace>
+
+            Test(workspace)
+        End Sub
+
+        <WpfFact, Trait(Traits.Feature, Traits.Features.GoToDefinition)>
+        Public Sub TestVisualBasicGoToOnContinueDoWhileLoop_Do()
+            Dim workspace =
+<Workspace>
+    <Project Language="Visual Basic" CommonReferences="true">
+        <Document>
+Class C
+    Sub M()
+        [||]Do While True
+             Continue Do$$
+        Loop
+    End Sub
+End Class
+        </Document>
+    </Project>
+</Workspace>
+
+            Test(workspace)
+        End Sub
+
+        <WpfFact, Trait(Traits.Feature, Traits.Features.GoToDefinition)>
+        Public Sub TestVisualBasicGoToOnExitDoWhileLoop_Do()
```

Comment:
nit, but it would be apprecaited.  please break the C# and Vb tsts in to separate test files.

Reply:
Good idea. Done!